### ISSUE TITLE
clang -Werror fails with -Wshift-negative-value on OCaml 4.02.3

### ISSUE THREAD

**Issue by djs55** [dsheets/ocaml-unix-unistd#11]
Xcode 7.3 has an updated clang which highlights undefined behaviour (left shift of a negative number, in this case -1) which breaks the build.

There's an upstream fix here: https://github.com/ocaml/ocaml/commit/24c118d7.diff

In case you want a workaround, we would some positive value instead of -1 for these constants?

```
lib/unix_unistd_stubs.c:73:49: error: shifting a negative signed value is undefined [-Werror,-Wshift-negative-value]
CAMLprim value unix_unistd_seek_data() { return Val_int(SEEK_DATA); }
                                                ^~~~~~~~~~~~~~~~~~
/Users/rneugeba/src/go/src/github.com/docker/pinata/_cache/opam/4.02.3/lib/ocaml/caml/mlvalues.h:75:20: note: expanded from macro 'Val_int'
#define Val_int(x) Val_long(x)
                   ^~~~~~~~~~~
/Users/rneugeba/src/go/src/github.com/docker/pinata/_cache/opam/4.02.3/lib/ocaml/caml/mlvalues.h:71:39: note: expanded from macro 'Val_long'
#define Val_long(x)     (((intnat)(x) << 1) + 1)
                          ~~~~~~~~~~~ ^
lib/unix_unistd_stubs.c:74:49: error: shifting a negative signed value is undefined [-Werror,-Wshift-negative-value]
CAMLprim value unix_unistd_seek_hole() { return Val_int(SEEK_HOLE); }
                                                ^~~~~~~~~~~~~~~~~~
/Users/rneugeba/src/go/src/github.com/docker/pinata/_cache/opam/4.02.3/lib/ocaml/caml/mlvalues.h:75:20: note: expanded from macro 'Val_int'
#define Val_int(x) Val_long(x)
                   ^~~~~~~~~~~
/Users/rneugeba/src/go/src/github.com/docker/pinata/_cache/opam/4.02.3/lib/ocaml/caml/mlvalues.h:71:39: note: expanded from macro 'Val_long'
#define Val_long(x)     (((intnat)(x) << 1) + 1)
                          ~~~~~~~~~~~ ^
2 errors generated.
make: *** [_build/lib/unix_unistd_stubs.o] Error 1
```

We should probably also disable `-Werror` in release builds.

**Comment by dsheets** [comment]
This has been fixed in OCaml 4.03.0 and the library hasn't yet been released so the only fix will be ensuring `-Werror` is off for releases.

**Comment by dsheets** [comment]
I don't believe this is still applicable now that the bindings are generated and 4.03.0 is more widespread.
